custodia/ipa/certreq.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (C) 2017  Custodia Project Contributors - see LICENSE file
+"""FreeIPA cert request store
+"""
+import datetime
+import textwrap
+
+from cryptography import x509
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.x509 import oid
+
+from ipalib.krb_utils import krb5_format_service_principal_name
+
+import six
+
+from custodia.plugin import CSStore, CSStoreError, PluginOption, REQUIRED
+from .interface import IPAInterface
+
+
+TLS_SERVERAUTH = oid.ObjectIdentifier('2.5.29.37.1')
+
+TEMPLATE = textwrap.dedent("""\
+    Issuer: {issuer}
+    Subject: {subject}
+    Serial Number: {cert.serial_number}
+    Validity:
+        Not Before: {cert.not_valid_before}
+        Not After: {cert.not_valid_after}
+    {pem}\
+""")
+
+
+class CSRBuilder(object):
+    def __init__(self, plugin, backend=None):
+        if backend is None:
+            self.backend = default_backend()
+        else:
+            self.backend = backend
+        self.plugin = plugin
+
+    def gen_private(self):
+        return rsa.generate_private_key(
+            public_exponent=65537,
```

Comment:
Maybe something like CHOSEN_SMALL_FERMAT_PRIME would fit, but I am not sure it would give more clarity or more obscurity ...